src/textual/debounce.py

```diff
@@ -0,0 +1,24 @@
+import asyncio
+from functools import wraps
+from typing import Callable, Optional
+
+from .app import App
+from .timer import Timer as TimerClass
+
+
+def debounce(timeout: float) -> Callable:
```

Comment:
Are you up for a challenge? Can we use the new `ParamSpec` to type the decorator? Without it the typing of a decorated function is lost.

Here's a blog post on the subject:

https://sobolevn.me/2021/12/paramspec-guide

Reply:
I think it's about time you add mypy to your CI. pre-commit.ci might be the best option!

Reply:
I've added an `ImportError` check as `ParamSpec` is >3.10 only.